iib/web/api_v1.py

```diff
@@ -314,23 +314,33 @@ def get_builds():
     return flask.jsonify(response)
 
 
-@api_v1.route('/healthcheck')
-def get_healthcheck():
+@api_v1.route("/healthcheck/details", methods=["GET"])
+def get_status():
     """
-    Respond to a health check.
+    Return status of IIB workers and services that IIB depends on.
 
+    :return: json object representing the status of the services.
     :rtype: flask.Response
-    :return: json object representing the health of IIB
-    :raises IIBError: if the database connection fails
     """
-    # Test DB connection
+    return flask.jsonify(status())
+
+
+@api_v1.route("/healthcheck", methods=["GET"])
+def get_status_short():
+    """
+    Return 200 if all workers and services seem to be ok, 503 otherwise.
+
+    :rtype: flask.Response
+    :return: json object representing the status of IIB services.
+    :raise IIBError: if a problem is found while retrieving the status.
+    """
     try:
-        db.engine.execute(text('SELECT 1'))
-    except Exception:
-        flask.current_app.logger.exception('DB test failed.')
-        raise IIBError('Database health check failed.')
+        status(short=True)
+        retval = {"status": "Health check OK"}
+    except IIBError as error:
+        retval = {"status": "Health check failed", "error": str(error)}
 
-    return flask.jsonify({'status': 'Health check OK'})
+    return flask.jsonify(retval), 200 if retval["status"] == "Health check OK" else 503
```

Comment:
It does return error. For example, in case db conncection failed 

`"status": "Health check failed", "error": "DATABASE unavailable: Database connection failed"`

may I ask what made you think it won't return any response text?

Reply:
Oh, my bad! I had written the line as:
```python
    return (flask.jsonify(retval), 200) if retval["status"] == "Health check OK" else 503
```

When in fact, it's processed like this (added parenthesis for clarity):
```python
return flask.jsonify(retval), (200 if retval["status"] == "Health check OK" else 503)
```

This doesn't change my comment about the conditional being brittle.

Reply:
I'll modify it! :)